src/editor/components/EditorElement.tsx

```diff
@@ -48,29 +56,76 @@ export const EditorElement: React.FC<EditorElementProps> = ({
   const elementSchema = elementSchemaPath
     ? getFromPath(schema, elementSchemaPath)
     : undefined;
+
+  const [openConfirmRemoveDialog, setOpenConfirmRemoveDialog] = React.useState(
+    false
+  );
   const [{ isDragging }, drag] = useDrag({
     item: DndItems.moveUISchemaElement(wrappedElement, elementSchema),
     collect: (monitor) => ({
       isDragging: !!monitor.isDragging(),
     }),
   });
   const classes = useEditorElementStyles({ isDragging });
+  const uiPath = getUISchemaPath(wrappedElement);
+  const dispatch = useDispatch();
+
   return (
     <Grid
       item
       container
       wrap='nowrap'
       direction='column'
+      data-cy={`editorElement-${uiPath}`}
       className={classes.editorElement}
       ref={drag}
-      xs
     >
       <Grid
         item
-        key={`${getUISchemaPath(wrappedElement)}-${0}-icon`}
-        className={classes.iconGridItem}
+        container
+        direction='row'
+        wrap='nowrap'
+        className={classes.elementHeader}
+        data-cy={`editorElement-${uiPath}-header`}
       >
-        {React.createElement(getIconForUISchemaType(wrappedElement.type), {})}
+        <Grid item container alignItems='center' xs>
+          <UISchemaIcon type={wrappedElement.type} />
+        </Grid>
+        <Grid
+          item
+          container
+          className={classes.elementControls}
+          justify='flex-end'
+          alignItems='center'
+          xs
+        >
+          <IconButton
+            data-cy={`editorElement-${uiPath}-removeButton`}
+            size='small'
+            onClick={() => {
+              if (
+                !Array.isArray((wrappedElement as Layout).elements) ||
```

Comment:
I would suggest to move this to a helper function something like `isEmptyLayout` this will make the code better readable